nms/app/packages/magmalte/app/state/lte/EquipmentState.js

```diff
@@ -435,3 +441,180 @@ export async function RunGatewayCommands(props: GatewayCommandProps) {
       }
   }
 }
+
+/**************************** Gateway Pools State **********************************/
+
+export async function FetchGatewayPools(props: FetchProps) {
+  const {networkId, id, enqueueSnackbar} = props;
+  if (id !== undefined && id !== null) {
+    try {
+      const gatewayPool = await MagmaV1API.getLteByNetworkIdGatewayPoolsByGatewayPoolId(
+        {
+          networkId: networkId,
+          gatewayPoolId: id,
+        },
+      );
+      return gatewayPool;
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching gateway information', {
+        variant: 'error',
+      });
+    }
+  } else {
+    try {
+      return await MagmaV1API.getLteByNetworkIdGatewayPools({
+        networkId: networkId,
+      });
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching gateway pools information', {
+        variant: 'error',
+      });
+    }
+  }
+}
+type GatewayPoolsStateProps = {
+  networkId: network_id,
+  gatewayPools: {[string]: gatewayPoolsStateType},
+  setGatewayPools: ({[string]: gatewayPoolsStateType}) => void,
+  gatewayId?: string,
+  key: gateway_pool_id,
+  value?: mutable_cellular_gateway_pool,
+  resources?: Array<GatewayPoolRecordsType>,
+};
+export async function SetGatewayPoolsState(props: GatewayPoolsStateProps) {
+  const {
+    networkId,
+    gatewayPools,
+    setGatewayPools,
+    key,
+    value,
+    resources,
+  } = props;
+  // Add gateways to gateway pool
+  if (resources != null) {
```

Comment:
minor nit: Setting resources seems pretty involved. Can you move it to a different function for readability. Also we could perhaps create it under a different method than setState.